craft_providers/bases/almalinux.py

```diff
@@ -83,7 +83,7 @@ def __init__(
         cache_path: Optional[pathlib.Path] = None,
     ) -> None:
         self._cache_path = cache_path
-        self.alias: AlmaLinuxBaseAlias = alias
```

Comment:
This wasn't matching the type in the [parent class](https://github.com/canonical/craft-providers/blob/896f58f7b98acf1d592701528e06d955c0762b2b/craft_providers/base.py#L99), even though `AlmaLinuxBaseAlias` is a subclass of `enum`.

```
/home/developer/dev/craft-providers/craft_providers/bases/almalinux.py
  /home/developer/dev/craft-providers/craft_providers/bases/almalinux.py:86:14 - error: "alias" overrides symbol of same name in class "Base"
    Variable is mutable so its type is invariant
      Override type "AlmaLinuxBaseAlias" is not the same as base type "Enum" (reportIncompatibleVariableOverride)
```

I think I did the right thing here, but I'm not sure.

Reply:
I wonder why pyright thinks an Enum of a string is mutable?  I suspect this is a variant of https://github.com/microsoft/pyright/issues/5993

Here's a minimal reproducer:

```
import enum
class Eee(enum.Enum):
    EEE = "eeeeeeee"

class Parent:
    enum_choice: enum.Enum


class Child(Parent):
    def __init__(self, e: Eee):
        self.enum_choice: Eee = e
```

mypy doesn't mark this as an issue, even with strict mode on.

I've [reported a bug to the pyright devs](https://github.com/microsoft/pyright/issues/6750) to see what they think since the changelogs don't include anything that jump out at me indicating this change.